packages/amplify-graphql-auth-transformer/src/__tests__/auth-operations.test.ts

```diff
@@ -351,3 +366,157 @@ test('can update on model without delete', () => {
+      name: 'owner, oidc, implicit owner field',
+      schema: '{allow: owner, provider: oidc, identityClaim: "sub", operations: [<OPERATION>]}',
+      auth: oidcAuthEntry,
+    },
+
+    {
+      name: 'owner, oidc, explicit owner field',
+      schema: '{allow: owner, provider: oidc, ownerField: "customOwner", identityClaim: "sub", operations: [<OPERATION>]}',
+      auth: oidcAuthEntry,
+    },
+
+    {
+      name: 'static groups, userPools, implicit groups field',
+      schema: '{allow: groups, groups: ["Admin", "Dev"], operations: [<OPERATION>]}',
+      auth: userPoolAuthEntry,
+    },
+
+    {
+      name: 'static groups, oidc, implicit groups field',
+      schema: '{allow: groups, groups: ["Admin", "Dev"], provider: oidc, groupClaim: "groups", operations: [<OPERATION>]}',
+      auth: oidcAuthEntry,
+    },
+
+    {
+      name: 'dynamic groups, userPools',
+      schema: '{allow: groups, groupsField: "customGroups", operations: [<OPERATION>]}',
+      auth: userPoolAuthEntry,
+    },
+
+    {
+      name: 'dynamic groups, oidc',
+      schema: '{allow: groups, groupsField: "customGroups", provider: oidc, groupClaim: "groups", operations: [<OPERATION>]}',
+      auth: oidcAuthEntry,
+    },
+
+    { name: 'custom', schema: '{allow: custom, operations: [<OPERATION>]}', auth: lambdaAuthEntry },
+  ];
+
+  const ddbSchemaTemplate = /* GraphQL */ `
+    type Post @model @auth(rules: [ <AUTH_RULE> ]) {
+      id: ID!
+      description: String
+    }
+  `;
+
+  const sqlSchemaTemplate = /* GraphQL */ `
+    type Post @model @auth(rules: [ <AUTH_RULE> ]) {
+      id: ID! @primaryKey
+      description: String
+    }
+  `;
+
+  for (const strategy of dataSourceStrategies) {
+    for (const rule of rules) {
+      for (const operation of operations) {
+        const strategyName = isSqlStrategy(strategy)
+          ? strategy.name
+          : isDefaultDynamoDbModelDataSourceStrategy(strategy)
+          ? 'CFn-managed DDB'
+          : 'Amplify-managed DDB';
+
+        const schemaTemplate = isSqlStrategy(strategy) ? sqlSchemaTemplate : ddbSchemaTemplate;
+
+        const transformers: TransformerPluginProvider[] = [new ModelTransformer(), new AuthTransformer()];
+        if (isSqlStrategy(strategy)) {
+          transformers.push(new PrimaryKeyTransformer());
+        }
+
+        test(`${strategyName} - ${rule.name} - ${operation}`, () => {
+          const resolvedAuthRule = rule.schema.replace('<OPERATION>', operation);
+          const schema = schemaTemplate.replace('<AUTH_RULE>', resolvedAuthRule);
+          const resolverConfig: ResolverConfig | undefined = operation === 'sync' ? { project: syncConfig } : undefined;
+
+          const out = testTransform({
+            schema,
+            authConfig: { defaultAuthentication: rule.auth, additionalAuthenticationProviders: [] },
+            transformers,
+            resolverConfig,
+            dataSourceStrategies: constructDataSourceStrategies(schema, strategy),
+          });
```

Comment:
`sync` and `search` are not supported for SQL datasource strategies. But the fact that the transformation succeeds here probably means we're missing those validations.